### PR TITLE
Instructor copy session: check for empty session name before submitting #5955

### LINKED ISSUE #5955: Instructor copy session: check for empty session name before submitting
Prevent submitting this form (and point out the problem) if the session name is empty.
image

The current one not only submits the form with the empty value, but also sends the user to a course different from the one he was using before.

### PR DESCRIPTION
**Pull request by rclakmal**
Fixes #5955

21hpcmin7x

### CODE REVIEW

**Comment by damithc** [comment]
UI looks fine. :+1:

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 1]
```diff
@@ -1011,5 +1011,24 @@ public boolean isElementInViewport(String id) {
...
     }
+    
+    /**
+     * @param elementId Id of the element
+     * @param targetClass className
+     * @return {@code true} if the element has the given class
+     */
+    public boolean isElementHasClass(String elementId, String targetClass) {
+        try {
+            String classNames = browser.driver.findElement(By.id(elementId)).getAttribute("class");
```
Comment (YongJieYongJie) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: I think you can use `By.cssSelector("#" + elementId + "." + targetClass)`

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 2]
```diff
@@ -1011,5 +1011,24 @@ public boolean isElementInViewport(String id) {
         String script = "return isWithinView(document.getElementById('" + id + "'));";
         return (boolean) jsExecutor.executeScript(script);
     }
+    
+    /**
+     * @param elementId Id of the element
+     * @param targetClass className
+     * @return {@code true} if the element has the given class
+     */
+    public boolean isElementHasClass(String elementId, String targetClass) {
```
Comment (YongJieYongJie) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: perhaps put this method near `isElementVisible()`

**Comment by YongJieYongJie** [comment]
@rclakmal Some changes requested

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 3]
```diff
@@ -687,6 +687,20 @@ public boolean isElementVisible(By by) {
...
+     * @param elementId
+     *            Id of the element
+     * @param targetClass
+     *            className
+     * @return {@code true} if there exists an element with the given id and
+     *         class name.
+     */
+    public boolean isElementHasClass(String elementId, String targetClass) {
+        List<WebElement> elementsMatched = browser.driver
+                .findElements(By.cssSelector("#" + elementId + "." + targetClass));
```
Comment (YongJieYongJie) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: if we break the line after the `=` sign, will it still fit within 125 char?

File: src/test/java/teammates/test/pageobjects/AppPage.java [hunk 4]
```diff
@@ -1001,5 +1015,5 @@ public boolean isElementInViewport(String id) {
         String script = "return isWithinView(document.getElementById('" + id + "'));";
         return (boolean) jsExecutor.executeScript(script);
     }
-
+
```
Comment (YongJieYongJie) [inline src/test/java/teammates/test/pageobjects/AppPage.java]: Try to avoid whitespace changes if possible

**Comment by YongJieYongJie** [comment]
@rclakmal Added some final stylistic comments

**Comment by rclakmal** [comment]
@YongJieYongJie Thanks for comments. Taken cared!

**Comment by YongJieYongJie** [comment]
@rclakmal Looks good!
(However, I think the branch name could have been more specific.)

**Pull request merged**